graph/telemetry/istio/appender/security_policy.go

```diff
@@ -188,10 +194,10 @@ func applySecurityPolicy(trafficMap graph.TrafficMap, securityPolicyMap map[stri
 	}
 }
 
-func (a SecurityPolicyAppender) addPrincipal(principalMap map[string]map[graph.MetadataKey]string, sourceNs, sourceSvc, sourceWl, sourceApp, sourceVer, sourcePrincipal, destSvcNs, destSvc, destWlNs, destWl, destApp, destVer, destPrincipal string) {
-	sourceId, _ := graph.Id(sourceNs, sourceSvc, sourceNs, sourceWl, sourceApp, sourceVer, a.GraphType)
-	destId, _ := graph.Id(destSvcNs, destSvc, destWlNs, destWl, destApp, destVer, a.GraphType)
-	key := fmt.Sprintf("%s %s", sourceId, destId)
+func (a SecurityPolicyAppender) addPrincipal(principalMap map[string]map[graph.MetadataKey]string, sourceCluster, sourceNs, sourceSvc, sourceWl, sourceApp, sourceVer, sourcePrincipal, destCluster, destSvcNs, destSvc, destWlNs, destWl, destApp, destVer, destPrincipal string) {
+	sourceID, _ := graph.Id(sourceCluster, sourceNs, sourceSvc, sourceNs, sourceWl, sourceApp, sourceVer, a.GraphType)
+	destID, _ := graph.Id(destCluster, destSvcNs, destSvc, destWlNs, destWl, destApp, destVer, a.GraphType)
+	key := fmt.Sprintf("%s %s", sourceID, destID)
```

Comment:
I see these lines repeated in other appenders. Not sure if they are a candidate to be extracted to a func.
I don't want a refactor on this PR. But I'm wondering if this deserves a refactor on a later PR.

Reply:
Yes, there are a few repetitive things in the appenders and I suppose it could be consolidated to avoid occasional updates in multiple places.  But to be honest I don't expect to do that consolidation.  As is the code is clear, and gives some flexibility if we do have minor differences, as the appenders handle distinct things.